--- make-3.81/job.c.orig	2006-03-20 03:03:04.000000000 +0000
+++ make-3.81/job.c	2014-02-20 23:34:15.094679900 +0000
@@ -2307,6 +2307,12 @@
                              "login", "logout", "read", "readonly", "set",
                              "shift", "switch", "test", "times", "trap",
                              "umask", "wait", "while", 0 };
+# ifdef HAVE_DOS_PATHS
+  /* This is required if the MSYS/Cygwin ports (which do not define
+     WINDOWS32) are compiled with HAVE_DOS_PATHS defined, which uses
+     sh_chars_sh[] directly (see below).  */
+  static char *sh_chars_sh = sh_chars;
+# endif  /* HAVE_DOS_PATHS */
 #endif
   register int i;
   register char *p;
