From f3af5531fa4299171408d7318339a92a4f10ff10 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 15 Jul 2014 21:50:30 +0100
Subject: [PATCH 38/38] pretend native is cross

---
 scripts/crosstool-NG.sh.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/crosstool-NG.sh.in b/scripts/crosstool-NG.sh.in
index b3108c5..3b504b5 100644
--- a/scripts/crosstool-NG.sh.in
+++ b/scripts/crosstool-NG.sh.in
@@ -367,7 +367,7 @@ if [ -z "${CT_RESTART}" ]; then
 
     # Prepare mangling patterns to later modify BUILD and HOST (see below)
     case "${CT_TOOLCHAIN_TYPE}" in
-        cross)
+        cross|native)
             # A cross-compiler runs on the same machine it is built on
             CT_HOST="${CT_BUILD}"
             build_mangle="build_"
@@ -510,7 +510,7 @@ if [ -z "${CT_RESTART}" ]; then
     # Of course, neither cross-native nor canadian can run on BUILD,
     # so don't add those PATHs in this case...
     case "${CT_TOOLCHAIN_TYPE}" in
-        cross)  export PATH="${CT_PREFIX_DIR}/bin:${CT_BUILDTOOLS_PREFIX_DIR}/bin:${PATH}";;
+        cross|native)  export PATH="${CT_PREFIX_DIR}/bin:${CT_BUILDTOOLS_PREFIX_DIR}/bin:${PATH}";;
         canadian) export PATH="${CT_BUILDTOOLS_PREFIX_DIR}/bin:${PATH}";;
         *)  ;;
     esac
-- 
2.1.3

